server/supernova/CMakeLists.txt

```diff
@@ -104,31 +104,57 @@ else()
   endif()
 endif()
 
-if(APPLE OR MSYS)
+if(AUDIOAPI STREQUAL "default")
+	if(APPLE)
+		set(AUDIOAPI portaudio)
+	elseif(WIN32)
+		set(AUDIOAPI portaudio)
+	else()
+		set(AUDIOAPI jack)
+	endif(APPLE)
+endif()
+
+if(NOT AUDIOAPI MATCHES "^(jack|coreaudio|portaudio)$")
+	message(FATAL_ERROR "Unrecognised audio API: ${AUDIOAPI}")
+endif()
+
+if(AUDIOAPI STREQUAL "coreaudio")
+	set(AUDIOAPI portaudio)
+  message(STATUS "Supernova: coreaudio API not implemented, using ${AUDIOAPI} instead")
```

Comment:
if it's not supported, i think this should be a fatal error rather than defaulting to something the user may not want. perhaps we need a separate CMake option for supernova audio api?